### ISSUE TITLE
Replication decreases query performance

### ISSUE THREAD

**Issue by benjaminrigaud** [crate/crate#2933]
When I configure replication (I tried with 1), all my queries are far slower.

I have a cluster of 4 machines with 2 ssd, no load and launching one query at a time.
My database has 4 shards and is partitionned by month (on 3 months) with 180M documents following this schema:

```
id long 
col_1 integer 
col_2 integer 
col_3 integer 
col_4 integer 
col_5 integer 
col_6 integer 
object_id integer 
object2_id integer 
object3_id integer 
object4_id integer 
start_time timestamp 
start_time__yearmonth timestamp 
col_7 boolean
```

A query such as:

```sql
SELECT COUNT(*), AVG(col_1), start_time__yearmonth from events WHERE start_time__yearmonth = '201510' GROUP BY start_time__yearmonth;
```

Is 2-3 times slower (went from 3s to 7-9s).

I tried crate 0.51.7 and 0.52.2.

**Comment by kovrus** [comment]
@benjaminrigaud we are currently trying to reproduce the reported behavior.

**Comment by benjaminrigaud** [comment]
Am using Ubuntu 14.04 and all communications are going over openVPN.

Similar queries with an elasticsearch client or kibana are working fine (2-3s).

**Comment by kovrus** [comment]
Hi @benjaminrigaud, the fix is on master now. I'll let you know when it gets to the testing release.

**Comment by benjaminrigaud** [comment]
@kovrus, great! Thank you.

**Comment by joemoe** [comment]
Hi @benjaminrigaud, the fix is now in 53.1. https://github.com/crate/crate/releases/tag/0.53.1 
We're looking forward to your feedback.

**Comment by benjaminrigaud** [comment]
I just tested on 4 typical requests: it looks all good. Thank you!
